### PR TITLE
allow dots at end of keys

### LINKED ISSUE #7960: Bug: Lang strings not work if that string ends with the dot (.) and at least two levels deep.
### PHP Version

8.0

### CodeIgniter4 Version

4.4.1

### CodeIgniter4 Installation Method

Manual (zip or tar.gz)

### Which operating systems have you tested for this bug?

Windows

### Which server did you use?

cli-server (PHP built-in webserver)

### Database

PostgreSQL 10

### What happened?

Test.php (Lang string file).

```php
<?php

return [
    'level_one' => [
        'level_two' => [
            'The string with the dot at the end.'
                => 'The string with the dot at the end.'
        ]
    ]
];

```

In the view file.

```php
<?= lang('Test.level_one.level_two.The string with the dot at the end.') ?>
```

The output.

```html
Test.level_one.level_two.The string with the dot at the end.
```

The expected output.
```html
The string with the dot at the end.
```

This only happen when the lang string key and the value two levels deep in the lang array and has the dot (.) at the end of lang key. Without the dot (.) it's works fine. But this issue only happening when the lang array element at least two levels deep.

### Steps to Reproduce

Info given

### Expected Output

Info given

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by paulbalandan**
**Description**
Fixes #7960 

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [x] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: tests/_support/Language/ab-CD/Allin.php [hunk 1]
```diff
@@ -10,8 +10,19 @@
...
-    'one' => 'Pyramid of Giza',
-    'tre' => 'Colossus of Rhodes',
-    'fiv' => 'Temple of Artemis',
-    'sev' => 'Hanging Gardens of Babylon',
+    'one'  => 'Pyramid of Giza',
+    'tre'  => 'Colossus of Rhodes',
+    'fiv'  => 'Temple of Artemis',
+    'sev'  => 'Hanging Gardens of Babylon',
+    'file' => [
+        'with\.' => [
```
Comment (kenjis) [inline tests/_support/Language/ab-CD/Allin.php:18]: Why does `'with\.'` need `\`?

In dot array syntax, keys can contain a dot without escaping.
https://codeigniter4.github.io/CodeIgniter4/helpers/array_helper.html#dot_array_search

File: tests/system/Language/LanguageTest.php [hunk 2]
```diff
@@ -356,4 +356,13 @@ public function testLanguageKeySeparatedByDot(): void
         $this->assertSame('The fieldname field is very short.', $this->lang->getLine('Foo.bar.min_length1', ['field' => 'fieldname']));
         $this->assertSame('The fieldname field is very short.', $this->lang->getLine('Foo.baz.min_length3.short', ['field' => 'fieldname']));
     }
+
+    public function testLanguageKeyWithDotAsNonMetacharacter(): void
+    {
+        $this->lang = Services::language('ab-CD', false);
+
+        $this->assertSame('File with dot.', $this->lang->getLine('Allin.file.with\..dot'));
```
Comment (kenjis) [inline tests/system/Language/LanguageTest.php:364]: There is probably no such use case.

File: tests/system/Language/LanguageTest.php [hunk 3]
```diff
@@ -356,4 +356,13 @@ public function testLanguageKeySeparatedByDot(): void
...
+        $this->assertSame('The string with the dot at the end.', $this->lang->getLine('Allin.level_one.The string with the dot at the end\.'));
```
Comment (kenjis) [inline tests/system/Language/LanguageTest.php:365]: I would like to write as `'getLine('Allin.level_one.The string with the dot at the end.')`.

**Review by michalsn — COMMENTED** [review]
I'm not sure about this one. The examples in the user guide indicate that the language files should contain a key and a string for that key. The key was never designed to hold a complete sentence.

**Comment by kenjis** [comment]
I'm still not sure we accept this change.
The key was never designed to hold a complete sentence.
Does it make sense to allow the inclusion of the last dot as an exception?

**Comment by michalsn** [comment]
Personally, I'm against it. I don't feel this is a good direction for our translations.

If someone wants to follow this pattern, there is `gettext` already which is way more advanced and easy to use with all the tooling around.

**Pull request closed without merging**